packages/vm/src/state/interface.ts

```diff
@@ -42,4 +43,6 @@ export interface EIP2929StateManager extends StateManager {
   isWarmedStorage(address: Buffer, slot: Buffer): boolean
   clearWarmedAccounts(): void
   generateAccessList?(addressesRemoved: Address[], addressesOnlyStorage: Address[]): AccessList
+  getProof?(address: Address, storageSlots: Buffer[]): Promise<Proof>
+  verifyProof?(proof: Proof): Promise<boolean>
```

Comment:
Have thought about the API of this a bit. For now I would make a pledge to not add this to the `StateManager` interface *at all* but rather treat this as some custom functionality we add to our own `StateManager` implementation.

The functionality is not used in the VM at all and this will rather confuse people and make life harder for implementers who want to do custom state manager implementations, if we bloat the interface more and more.

Instead we should add a dev comment (so: no public code function comment but rather only visible within the code) to both functions like:

```typescript
/*
 * This function is currently not used within the VM but exists as a utility function for people 
 * who want to use the StateManager out of the VM context. It is therefore not part of the 
 * StateManager interface.
 *
 * If at some point it gets necessary to use VM internally make sure add to the StateManager interface.
 */
 ```

Does this make sense?

Reply:
Yes, this makes sense. We should also create a standalone StateManager package, see #1599.

Reply:
Do you think we should create an extra "type" of StateManager which has these proof methods? Like we did with EIP2929 state manager - the state manager with proofs would inherit the EIP2929 functions.